test/common/router/route_fuzz_test.cc

```diff
@@ -48,22 +48,20 @@ cleanRouteConfig(envoy::config::route::v3::RouteConfiguration route_config) {
 bool validateMatcherConfig(const xds::type::matcher::v3::Matcher& matcher);
 
 bool validateOnMatchConfig(const xds::type::matcher::v3::Matcher::OnMatch& on_match) {
-  if (on_match.has_matcher() && !validateMatcherConfig(on_match.matcher())) {
-    return false;
-  }
-  // Only envoy...Route is allowed as typed_config here.
-  if (on_match.action().typed_config().type_url().find("envoy.config.route.v3.Route") ==
-      std::string::npos) {
-    return false;
-  }
-  envoy::config::route::v3::Route on_match_route_action_config;
-  MessageUtil::unpackTo(on_match.action().typed_config(), on_match_route_action_config);
-  ENVOY_LOG_MISC(trace, "typed_config of on_match.action is: {}",
-                 on_match_route_action_config.DebugString());
-  if (isUnsupportedRouteConfig(on_match_route_action_config)) {
-    return false;
+  if (on_match.has_matcher()) {
```

Comment:
I went for the switch-case way. I like that more, because without a default-alternative the clang-compiler will warn about new entries. And with treat all warnings as errors it will point the developer to the location he missed this way ensuring that he covers the missing cases.